### PR TITLE
common.sh: fix last timestamp is null bug in last_dmesg_timestamp fun…

### PR DESCRIPTION
**Pull request by xupengfe**
…ction

There is no issue when there is no space in last line dmesg time stamp like as below:
[11501.945229] xxxx
It could get correct time stamp "11501.945229".

If there is space in last line dmesg time stamp like as below: [ 8275.926781] xxxx
It could not get correct last line dmesg time stamp, and with null "" instead.

So correct the awk judgement to fix above last time stamp bug.

Signed-off-by: Pengfei Xu <pengfei.xu@intel.com>

### CODE REVIEW

**Comment by ysun** [comment]
Thank Pengfei's fix.

**@hanning0511  I noticed 3 legacy issues raised by the shell check. The variables BIN_OUTPUT, BIN_RET and BIN_DMESG can be used further in cases.
Should we export them here?**

```
    common/common.sh:420 - BIN_OUTPUT appears unused. Verify use (or export if used externally)., refer to https://github.com/koalaman/shellcheck/wiki/SC2034
    common/common.sh:421 - BIN_RET appears unused. Verify use (or export if used externally)., refer to https://github.com/koalaman/shellcheck/wiki/SC2034
    common/common.sh:423 - BIN_DMESG appears unused. Verify use (or export if used externally)., refer to https://github.com/koalaman/shellcheck/wiki/SC2034
```

**Comment by xupengfe** [comment]
> Thank Pengfei's fix.
> 
> **@hanning0511 I noticed 3 legacy issues raised by the shell check. The variables BIN_OUTPUT, BIN_RET and BIN_DMESG can be used further in cases. Should we export them here?**
> 
> ```
>     common/common.sh:420 - BIN_OUTPUT appears unused. Verify use (or export if used externally)., refer to https://github.com/koalaman/shellcheck/wiki/SC2034
>     common/common.sh:421 - BIN_RET appears unused. Verify use (or export if used externally)., refer to https://github.com/koalaman/shellcheck/wiki/SC2034
>     common/common.sh:423 - BIN_DMESG appears unused. Verify use (or export if used externally)., refer to https://github.com/koalaman/shellcheck/wiki/SC2034
> ```

In my test experience, we could use "BIN_OUTPUT, BIN_RET, BIN_DMESG" directly.
It's just a remind if the variable is useless,  but the shellcheck script doesn't know it's useful in other shell scripts.

Thanks!

**Comment by hanning0511** [comment]
> > Thank Pengfei's fix.
> > **@hanning0511 I noticed 3 legacy issues raised by the shell check. The variables BIN_OUTPUT, BIN_RET and BIN_DMESG can be used further in cases. Should we export them here?**
> > ```
> >     common/common.sh:420 - BIN_OUTPUT appears unused. Verify use (or export if used externally)., refer to https://github.com/koalaman/shellcheck/wiki/SC2034
> >     common/common.sh:421 - BIN_RET appears unused. Verify use (or export if used externally)., refer to https://github.com/koalaman/shellcheck/wiki/SC2034
> >     common/common.sh:423 - BIN_DMESG appears unused. Verify use (or export if used externally)., refer to https://github.com/koalaman/shellcheck/wiki/SC2034
> > ```
> 
> In my test experience, we could use "BIN_OUTPUT, BIN_RET, BIN_DMESG" directly. It's just a remind if the variable is useless, but the shellcheck script doesn't know it's useful in other shell scripts.
> 
> Thanks!

@xupengfe 
Notice that bin_output_dmesg() is create by you. IMO, it's better to put this logic in a test case script, rather than in common.sh. It's a bit redundant to put them in common.sh.

File: common/common.sh [hunk 1]
```diff
@@ -359,7 +359,7 @@ dmesg_pattern_check() {
 # Record last timestamp in dmesg and store the value in variable
 # LAST_DMESG_TIMESTAMP. The value is refered in function extract_case_dmesg.
 last_dmesg_timestamp() {
-  LAST_DMESG_TIMESTAMP=$(dmesg | tail -n1 | awk '{print $1}' | tr -d "[]")
+  LAST_DMESG_TIMESTAMP=$(dmesg | tail -n1 | awk -F "]" '{print $1}' | tr -d "[]")
```
Comment (hanning0511) [inline common/common.sh:362]: @xupengfe , can you help explain why we need to set "]" as field seperator explicitly?

Comment (xupengfe) [reply common/common.sh:362]: If there is no  -F "]",  awk will check the first space as  "field seperator"


If there is space in last line dmesg time stamp, it will got null time stamp, it's wrong.
Like below example,  if there is no  -F "]",  will get null time stamp.
[  275.926781] xxxx

Thanks!

Comment (hanning0511) [reply common/common.sh:362]: Got it. Thanks.

File: common/common.sh [hunk 2]
```diff
@@ -417,8 +417,8 @@ bin_output_dmesg() {
   last_dmesg_timestamp
   test_print_trc "$bin_name $parm"
   [[ -e "$bin_name" ]] || block_test "No $bin_name in lkvs, do you compile it?"
-  BIN_OUTPUT=$($bin_name $parm 2>/dev/null)
-  BIN_RET=$?
+  BIN_OUTPUT=$($bin_name "$parm" 2>/dev/null)
```
Comment (hanning0511) [inline common/common.sh:420]: Modification of this function doesn't appear to be directly related to fixing the timestamp bug. Better to put them in another commit or PR.

**Comment by xupengfe** [comment]
> > > Thank Pengfei's fix.
> > > **@hanning0511 I noticed 3 legacy issues raised by the shell check. The variables BIN_OUTPUT, BIN_RET and BIN_DMESG can be used further in cases. Should we export them here?**
> > > ```
> > >     common/common.sh:420 - BIN_OUTPUT appears unused. Verify use (or export if used externally)., refer to https://github.com/koalaman/shellcheck/wiki/SC2034
> > >     common/common.sh:421 - BIN_RET appears unused. Verify use (or export if used externally)., refer to https://github.com/koalaman/shellcheck/wiki/SC2034
> > >     common/common.sh:423 - BIN_DMESG appears unused. Verify use (or export if used externally)., refer to https://github.com/koalaman/shellcheck/wiki/SC2034
> > > ```
> > 
> > 
> > In my test experience, we could use "BIN_OUTPUT, BIN_RET, BIN_DMESG" directly. It's just a remind if the variable is useless, but the shellcheck script doesn't know it's useful in other shell scripts.
> > Thanks!
> 
> @xupengfe Notice that bin_output_dmesg() is create by you. IMO, it's better to put this logic in a test case script, rather than in common.sh. It's a bit redundant to put them in common.sh.

I want to get the common function that:   it helps to catch the  binary  output,  return value and the dmesg info when executed one binary or some scripts,  and we don't need to prepare the similar functions in each domain functions I think.

That's the purpose for it in common.sh.

Thanks!

File: common/common.sh (hunk 2 above)
Comment (xupengfe) [reply common/common.sh:420]: Make sense, I will put them in another commit.

Thanks!

**Comment by ysun** [comment]
> or export if used externally

@xupengfe  I understand the 3 variables are useful. as suggested in the shell check: "or export if used externally".
I'm not asking to remove them, but **export** them instead of just declaring them.

**Comment by hanning0511** [comment]
> > > > Thank Pengfei's fix.
> > > > **@hanning0511 I noticed 3 legacy issues raised by the shell check. The variables BIN_OUTPUT, BIN_RET and BIN_DMESG can be used further in cases. Should we export them here?**
> > > > ```
> > > >     common/common.sh:420 - BIN_OUTPUT appears unused. Verify use (or export if used externally)., refer to https://github.com/koalaman/shellcheck/wiki/SC2034
> > > >     common/common.sh:421 - BIN_RET appears unused. Verify use (or export if used externally)., refer to https://github.com/koalaman/shellcheck/wiki/SC2034
> > > >     common/common.sh:423 - BIN_DMESG appears unused. Verify use (or export if used externally)., refer to https://github.com/koalaman/shellcheck/wiki/SC2034
> > > > ```
> > > 
> > > 
> > > In my test experience, we could use "BIN_OUTPUT, BIN_RET, BIN_DMESG" directly. It's just a remind if the variable is useless, but the shellcheck script doesn't know it's useful in other shell scripts.
> > > Thanks!
> > 
> > 
> > @xupengfe Notice that bin_output_dmesg() is create by you. IMO, it's better to put this logic in a test case script, rather than in common.sh. It's a bit redundant to put them in common.sh.
> 
> I want to get the common function that: it helps to catch the binary output, return value and the dmesg info when executed one binary or some scripts, and we don't need to prepare the similar functions in each domain functions I think.
> 
> That's the purpose for it in common.sh.
> 
> Thanks!

Got it. Make sense. I think we may need to rename the function. At least for me, it's hard to guess what do the function do from it's name.

**Comment by xupengfe** [comment]
Hi Ning,
Sure,  thanks for improvement if any for binary  dmesg output function.

Thanks!

**Review by hanning0511 — APPROVED** [review]
